src/jit/decomposelongs.cpp

```diff
@@ -796,11 +799,9 @@ GenTree* DecomposeLongs::DecomposeInd(LIR::Use& use)
         addrBase->TypeGet(), addrBase->AsLclVarCommon()->GetLclNum(), BAD_IL_OFFSET);
     GenTreePtr addrHigh = new(m_compiler, GT_LEA) GenTreeAddrMode(TYP_REF, addrBaseHigh, nullptr, 0, genTypeSize(TYP_INT));
     GenTreePtr indHigh = new (m_compiler, GT_IND) GenTreeIndir(GT_IND, TYP_INT, addrHigh, nullptr);
-    
+
     // Insert the nodes into the block
-    m_blockRange.InsertAfter(addrBaseHigh, indLow);
-    m_blockRange.InsertAfter(addrHigh, addrBaseHigh);
-    m_blockRange.InsertAfter(indHigh, addrHigh);
+    m_blockRange.InsertAfter(LIR::SeqTree(m_compiler, indHigh), indLow);
```

Comment:
After some more thought, I've remembered that there's a deeper problem with calling `SetTreeSeq` on a tree that contains nodes tht are already in an LIR block: the implementation calls `fgSetTreeSeq`, which mutates `gtNext`/`gtPrev` for all nodes in the tree. This is very much not safe for nodes that are already in a block, as blindly resequencing such nodes may perform incorrect code motion. IIRC I even added a comment to this effect in `SetTreeSeq` as part of #24.